src/expression.js

```diff
@@ -234,19 +238,30 @@ pp.parseSubscripts = function(base, startPos, startLoc, noCalls) {
     } else if (this.eat(tt.bracketL)) {
       let node = this.startNodeAt(startPos, startLoc)
       node.object = base
-      node.property = this.parseExpression()
+      node.property = this.parseExpression(false, refContextDestructuringErrors)
       node.computed = true
       this.expect(tt.bracketR)
       base = this.finishNode(node, "MemberExpression")
     } else if (!noCalls && this.eat(tt.parenL)) {
+      let refDestructuringErrors = new DestructuringErrors
+      let exprList = this.parseExprList(tt.parenR, false, false, refDestructuringErrors)
+      if (maybeAsyncArrow && !this.canInsertSemicolon() && this.eat(tt.arrow)) {
```

Comment:
I don't think it is possible for `maybeAsyncArrow` to be true (which depends on `base.name == "async"`) and this branch, whose conditional uses `this.eat(tt.parentL)`, to also be taken, since both would be referring to the same token, which can't both be `async` and `(`. Am I missing something?